[WIP] Scalafmt on save

belamenso opened the pull request.

belamenso commented on src/main/scala/repro/overloadedmap.scala:
```diff
@@ -7,7 +7,7 @@ import NamedTuple.{NamedTuple, AnyNamedTuple}
 class Query2[A]():
   def map[B](f: Expr2.Ref[A, NExpr] => Expr2[B, NExpr]): Query2[B] = ???
-  def map[B <: AnyNamedTuple : Expr2.IsTupleOfExpr](f: Expr2.Ref[A, NExpr] => B): Query2[ NamedTuple.Map[B, Expr2.StripExpr2] ] = ???
+  def map[B <: AnyNamedTuple: Expr2.IsTupleOfExpr](f: Expr2.Ref[A, NExpr] => B): Query2[NamedTuple.Map[B, Expr2.StripExpr2]] = ???
```
I dislike it removing this space.

belamenso replied:
fixed by `spaces {
  beforeContextBoundColon = IfMultipleBounds
`

belamenso commented on src/main/scala/repro/overloadedmap.scala:
```diff
@@ -23,25 +23,21 @@ object Expr2:
   type IsTupleOfAgg[A <: AnyNamedTuple] = Tuple.Union[NamedTuple.DropNames[A]] <:< Expr2[?, ScalarExpr]
-  extension [A <: AnyNamedTuple : IsTupleOfAgg](x: A)
-    def toRow: AggProject[A] = ???
+  extension [A <: AnyNamedTuple: IsTupleOfAgg](x: A) def toRow: AggProject[A] = ???
```
I dislike it pushing extensions into a single line

aherlihy replied:
I agree, can we turn that rule off?

belamenso replied:
Yes, it's fixed now. But now it does
```
case class SelectQuery
  (
      project: QueryIRNode,
      from: Seq[RelationOp],
      where: Seq[QueryIRNode],
      overrideAlias: Option[String],
      ast: DatabaseAST[?]
  ) extends RelationOp:
```
instead of
```
case class SelectQuery(project: QueryIRNode,
                       from: Seq[RelationOp],
                       where: Seq[QueryIRNode],
                       overrideAlias: Option[String],
                       ast: DatabaseAST[?]) extends RelationOp:
```

I think these behaviors are governed by the same setting. I think that this is mostly acceptable.

belamenso commented:
It's also of course possible to use the IntelliJ formatter for Scala, which possibly has more features and is built-in. I don't know what the best practices for Scala projects are for this.

belamenso committed:
24.11.02 fix broken tests

right now on `sbt run test` it fails due to a misplaced file:
[error] java.sql.SQLException: java.sql.SQLException: IO Error: No files found that match the pattern "/home/bbi/ticklish/tyql/bench/data/tc/edge.csv"
[error] 	at org.duckdb.DuckDBPreparedStatement.prepare(DuckDBPreparedStatement.java:121)
[error] 	at org.duckdb.DuckDBPreparedStatement.execute(DuckDBPreparedStatement.java:195)
[error] 	at tyql.main$package$.main(main.scala:57)
[error] 	at tyql.main.main(main.scala:39)
[error] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[error] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[error] Caused by: java.sql.SQLException: IO Error: No files found that match the pattern "/home/bbi/ticklish/tyql/bench/data/tc/edge.csv"
[error] 	at org.duckdb.DuckDBNative.duckdb_jdbc_prepare(Native Method)
[error] 	at org.duckdb.DuckDBPreparedStatement.prepare(DuckDBPreparedStatement.java:115)
[error] 	at org.duckdb.DuckDBPreparedStatement.execute(DuckDBPreparedStatement.java:195)
[error] 	at tyql.main$package$.main(main.scala:57)
[error] 	at tyql.main.main(main.scala:39)
[error] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[error] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[error] stack trace is suppressed; run last Compile / run for the full output
[error] (Compile / run) java.sql.SQLException: java.sql.SQLException: IO Error: No files found that match the pattern "/home/bbi/ticklish/tyql/bench/data/tc/edge.csv"
[error] Total time: 1 s, completed 2 Nov 2024, 16:30:15

belamenso committed:
Merge pull request #2 from belamenso/24.11.02-fix-broken-tests

24.11.02 fix broken tests

belamenso commented:
I think the current version is workable

belamenso committed:
Merge branch 'aherlihy:main' into main

belamenso committed:
Scalafmt config

belamenso committed:
Format according to scalafmt
